### PR TITLE
Added info to CONTRIBUTING.md about automatic format checks

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

Addition to #3513

## Tasklist
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,6 +7,17 @@ You can add a :+1: emoji to the issue if you want to express interest in this.
...
 
 We use `clang-format` version `3.8` to consistently format the code base. There is a helper script under `scripts/format.sh`.
+The format is automatically checked by the `mason-linux-release` job of a Travis CI build.
+To save development time a local hook `.git/hooks/pre-push`
+```
+#!/bin/sh
+
+remote="$1"
+if [ x"$remote" = xorigin ] ; then
+    ./scripts/format.sh || (echo "Unstaged format changes" ; exit 1)
```
Comment (MoKob) [inline CONTRIBUTING.md]: really makes your life easier

**Comment by daniel-j-h** [comment]
Hm should we add the hook script into the scripts directory and add a note here on how to symlink to it? Or would that result in issues when we change the hook script and users jump around commits via git?

**Comment by oxidase** [comment]
@daniel-j-h i don't think it is ok from the security side. git hooks are too private and I would copy-paste either from CONTRIBUTING.md or from the file.

Just thinking about a small extension: `scripts/format.sh` changes all files, so i would like to add `git rev-list $branch --not --remotes=origin`  to change only pushed files.

**Comment by oxidase** [comment]
Ok, just after some attempts a :japanese_ogre: line
```
files=$(git diff --name-only $(git rev-list --reverse HEAD --not --remotes=$remote | head -1)~ HEAD | grep -e ".[ch]pp$")
```
to get the list of cpp/hpp files that will be pushed. Also `scripts/format.sh` needs changes to format only a list of files. 

I will add just a warning comment line that `scripts/format.sh` formats all local cpp/hpp. If you feel that formating of only pushed files is good, please create an improvement ticket.

**Pull request merged**